criu/page-xfer.c

```diff
@@ -561,9 +583,16 @@ static int page_server_check_parent(int sk, struct page_server_iov *pi)
 	if (ret < 0)
 		return -1;
 
-	if (write(sk, &ret, sizeof(ret)) != sizeof(ret)) {
-		pr_perror("Unable to send response");
-		return -1;
+	if (opts.tls_psk_key) {
```

Comment:
Can we introduce helpers for all those checks if (opts.tls_psk_key)?
I struggle to find a proper name, so it's up to you, but something like:
```
int __send(int sockfd, const void *buf, size_t len)
{
        size_t transferred = opts.tls_psk_key ? tls_send(buf, len) : write(sk, buf, len);

        return (transferred - len);
}
```
Same for recv

Reply:
>Can we introduce helpers for all those checks if (opts.tls_psk_key)?

That is a good idea, thanks! Another thing is that `tls_send()` does not currently pass the `flags` argument for `send()` (e.g. `MSG_DONTWAIT`). I will try to add this too.